### ISSUE TITLE
Hierarchy creates improper files

### ISSUE THREAD

**Issue by joneshf** [purescript/purescript#3012]
The `hierarchy` command is making some improper files:

```dot
➜  purescript-prelude git:(master) purs hierarchy src/Control/Monad.purs 
digraph Control_Monad {
  ProperName {runProperName = "Applicative"} -> ProperName {runProperName = "Monad"};
  ProperName {runProperName = "Bind"} -> ProperName {runProperName = "Monad"};
}
```

It produces this image if run through `dot`:

![control_monad](https://user-images.githubusercontent.com/1356417/28758128-ae396df0-7547-11e7-9c4c-e2a027cb98f7.png)

The problem stems from not unwrapping the `ProperName`.

PR impending.

**Comment by hdgarrood** [comment]
Oh, wow, this must have been broken for years - it was a really long time ago that we switched to derived Show instances! Thanks for the report!

**Comment by hdgarrood** [comment]
Fixed by #3013 .
